lib/services/database.dart

```diff
@@ -31,47 +31,49 @@ class Database {
-  //   NetworkHelper networkHelper =
-  //       NetworkHelper('${APIPath.getCasesTimeline()}');
+  Future<CaseTimeLine> getCasesTimeline() async {
+    NetworkHelper networkHelper =
+        NetworkHelper('${APIPath.getCasesTimeline()}');
 
-  //   return Cases.fromJson(await networkHelper.getData());
-  // }
+    return CaseTimeLine.fromJson(await networkHelper.getData());
+  }
 
-  // Future<dynamic> fetchRecord(
-  //     {@required int pageNumber, @required int limit}) async {
-  //   NetworkHelper networkHelper =
-  //       NetworkHelper('${APIPath.fetchRecord(pageNumber, limit)}');
+  Future<RecordList> fetchRecord(
+      {@required int pageNumber, @required int limit}) async {
+    NetworkHelper networkHelper =
+        NetworkHelper('${APIPath.fetchRecord(pageNumber, limit)}');
 
-  //   return networkHelper.getData();
-  // }
+    return RecordList.fromJson(await networkHelper.getData());
+  }
 
-  // Future<dynamic> fetchRecordByAge({@required int age}) async {
-  //   NetworkHelper networkHelper =
-  //       NetworkHelper('${APIPath.fetchRecordByAge(age)}');
+  Future<RecordList> fetchRecordByAge({@required int age}) async {
+    NetworkHelper networkHelper =
+        NetworkHelper('${APIPath.fetchRecordByAge(age)}');
 
-  //   return networkHelper.getData();
-  // }
+    return RecordList.fromJson(await networkHelper.getData());
+  }
 
-  // Future<dynamic> fetchRecordByAgeGroup(
-  //     {@required int minAge, @required int maxAge}) async {
-  //   NetworkHelper networkHelper =
-  //       NetworkHelper('${APIPath.fetchRecordByAgeGroup(minAge, maxAge)}');
+  Future<RecordList> fetchRecordByAgeGroup(
+      {@required int minAge, @required int maxAge}) async {
+    NetworkHelper networkHelper =
+        NetworkHelper('${APIPath.fetchRecordByAgeGroup(minAge, maxAge)}');
 
-  //   return networkHelper.getData();
-  // }
+    print(await networkHelper.getData());
 
-  // Future<dynamic> fetchRecordByMonth({@required int monthNumber}) async {
-  //   NetworkHelper networkHelper =
-  //       NetworkHelper('${APIPath.fetchRecordByAge(monthNumber)}');
+    return RecordList.fromJson(await networkHelper.getData());
+  }
+
+  Future<RecordList> fetchRecordByMonth({@required int monthNumber}) async {
+    NetworkHelper networkHelper =
+        NetworkHelper('${APIPath.fetchRecordByAge(monthNumber)}');
 
-  //   return networkHelper.getData();
-  // }
+    return RecordList.fromJson(await networkHelper.getData());
+  }
 
-  // Future<dynamic> fetchRecordByRegion({@required String region}) async {
-  //   NetworkHelper networkHelper =
-  //       NetworkHelper('${APIPath.fetchRecordByRegion(region)}');
+  Future<RecordList> fetchRecordByRegion({@required String region}) async {
+    NetworkHelper networkHelper =
+        NetworkHelper('${APIPath.fetchRecordByRegion(region)}');
 
-  //   return networkHelper.getData();
-  // }
+    return RecordList.fromJson(await networkHelper.getData());
+  }
```

Comment:
It would be much better if you will make an Interface/Abstract class for:
1. CaseTimeLine and RecordList.
2. Case and Record.

so that instead of using `Future\<RecordList\>` and `Future\<CaseTimeLine\>` you can just use `Future\<YourInterface/AbstractClassHere\>` instead.

Benefits:
1. Complies to [Open/Closed Principle](https://www.youtube.com/watch?v=-ptMtJAdj40)
2. Let's say you want to changed `RecordList` or `CaseTimeLine` to a different class don't have to modify this file ever again, because you just need to pass a class that implements `YourInterface/AbstractClassHere`.
3. Much easier to test because you only have to mock 1 class that implements `YourInterface/AbstractClassHere` instead of mocking individual files(`RecordList` & `CaseTimeLine` & etc.).